docs/madr/decisions/035-mesh-metric-policy.md

```diff
@@ -0,0 +1,171 @@
+  profile: minimal # one of minimal, default, full
+  regex: http2_act.* # only profile or regex can be defined
+```
+3. Should we also think about grouping the settings somehow, tls/port/path/tags/envoy are all “mesh/sidecar” metrics and 
+“aggregate” are application metrics, or maybe we should just rename `aggregate` to `otherContainers`?
+
+#### What other meshes do
+- Istio has [Telemetry](https://istio.io/latest/docs/reference/config/telemetry/#Metrics) policy that allows defining providers (our backends), overrides (that allows disabling / changing tags), reporting interval
+- LinkerD has only tutorials on how to [export metrics](https://linkerd.io/2.14/tasks/exporting-metrics/#proxy), and creating a service profile can enable per route metrics
+
+#### New API suggestion:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshMetrics
+metadata:
+  name: default
+  namespace: kuma-system
+spec:
+  targetRef:
+    kind: Mesh | MeshSubset | MeshService | MeshServiceSubset
+  default:
+    sidecar:
+      profile: minimal
+      regex: http2_act.* # only profile or regex can be defined
+      usedOnly: true
+    application:
+      - path: "/metrics/prometheus" # is name still needed since targetRef can just point to a name?
+        port: 8888
+        regex: http.*
+      - port: 8000 # default path is /metrics
+    backends:
+      - type: Prometheus
+        prometheus:
+          port: 5670
+          path: /metrics
+          tls:
+            mode: Disabled | ProvidedTLS | ActiveMTLSBackend
+      - type: OpenTelemetry
+        openTelemetry:
+          endpoint: otel-collector:4317
+```
+
+### Configure MADS based on policy data
+MADS server is responsible for aggregating information about dataplanes needed for Prometheus monitoring. 
+It now extracts information from the `Mesh` resource which proxy should be monitored and creates a `MonitoringAssignment` response.
+
+We need to change the source of our information about monitored proxies from `Mesh` to `MeshMetric` policy. We cannot treat 
+new policy as usual in this context, we should not merge policies, but create different `MonitoringAssignment` for each 
+defined policy, this is needed for scenarios like migrating to a new backend. For example, when we define two `MeshMetric` 
+policies that target the same data plane proxy, we should create two separate `MonitoringAssignment` responses. 
+We can achieve this by extending Prometheus integration, so that it sends backend name with requests to MADS API.
+
+### Pushing metrics from DP to OTEL collector
+In order to push metrics to the OTEL collector, and preserve current Prometheus metrics, we need to start using [OTEL metrics sdk](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric). 
+It allows us to export metrics in multiple formats, especially the ones that we are interested in, Prometheus and OTLP. 
+Thanks to this, we will be able to preserve our old functionality and add the possibility to push metrics to the OTEL collector.
```

Comment:
I think it's fine if the otel collector doesn't land in 2.6. It'd be great but worth having this lower in the priority list